flask_server.py

```diff
@@ -71,27 +71,42 @@ def index():
     cookie_path = settings["cookie_path"] if "cookie_path" in settings else "token.txt"
     folder_path = settings["folder_path"]  if "folder_path" in settings else "ig_media"
     if request.method == "POST":
-        amountScraped = int(request.form["amountToScrape"])
+        if not os.path.exists("token.txt"):
+            log_lines = get_log_file_list() 
+            return render_template('index.html', count_i= count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = False, logged_in = False)
+        if request.form["amountToScrape"].isnumeric():
+            amountScraped = int(request.form["amountToScrape"])
+        else:
+            log_lines = get_log_file_list() 
+            return render_template('index.html', count_i= count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = True, logged_in = True)
         mode = request.form["mode_dropdown"]
         base64_media = start_scrape(cookie_path, folder_path, amountScraped, mode)
         rendered_base64_media = render_base64_media(base64_media)
     log_lines = get_log_file_list()   
     if len(log_lines) > 0:
         count_u, count_i, count_v = get_stats_from_log_line(log_lines)      
-    return render_template('index.html', count_i = count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media)
+    return render_template('index.html', count_i = count_i, count_v = count_v, log_lines = log_lines, images = rendered_base64_media, empty_form = False, logged_in = True)
 
 @app.route("/settings/", methods=['GET','POST'])
 def settings():
+    credentials_error = False
     settings = get_settings()    # Gets the settings
     if request.method == "POST":
         updated_settings = settings
         for setting in request.form:
             if len(request.form[setting]) > 0:
                 updated_settings[setting] = request.form[setting]
         save_settings(updated_settings)
+        if request.form["username"] and request.form["password"] != "":
+            if not get_session_id(request.form["username"], request.form["password"]) == 0:
+                credentials_error = True
     cookie_path = settings["cookie_path"] if "cookie_path" in settings else "token.txt"
     folder_path = settings["folder_path"]  if "folder_path" in settings else "ig_media"
-    return render_template("settings.html", folder_path = folder_path, cookie_path = cookie_path)
+    if os.path.exists("token.txt"):
+        logged_in = True
+    else:
+        logged_in = False
+    return render_template("settings.html", folder_path = folder_path, cookie_path = cookie_path, logged_in = logged_in, credentials_error = credentials_error)
```

Comment:
Let's not add a param for every single "warning". Create a single one as a dict which contains the others eg:

render_template(..., disclaimer = { logged_in: false, credential_errors : true })